torchtext/transforms.py

```diff
@@ -236,187 +236,16 @@ def __init__(
-            while i < len(token_list):
-                j = self._list_str_index(token_list, first, i)
-                if j != -1:
-                    new_token_list.extend(token_list[i:j])
-                    i = j
-                else:
-                    new_token_list.extend(token_list[i:])
-                    break
-
-                if (
-                    token_list[i] == first
-                    and i < len(token_list) - 1
-                    and token_list[i + 1] == second
-                ):
-                    new_token_list.append(first + second)
-                    i += 2
-                else:
-                    new_token_list.append(token_list[i])
-                    i += 1
-
-            token_list = list(new_token_list)
-            if len(token_list) == 1:
-                break
-            else:
-                pairs = self._get_pairs(token_list)
-
-        return token_list
-
-    @torch.jit.export
-    def _byte_encode(self, token: str) -> List[str]:
-        """Encode byte into an unicode character.
-
-        Equivalent to: (self.byte_encoder[b] for b in token.encode('utf-8')
-        """
-        encoded: List[str] = []
-        if torch.jit.is_scripting():
-            for b in token:
-                encoded.append(self.byte_encoder[ord(b)])
-        else:
-            encoded = [self.byte_encoder[b] for b in token.encode('utf-8')]
-        return encoded
-
-    @torch.jit.export
-    def _regex(self, text: str) -> List[str]:
-        r"""Return a list of tokens, split by regular expression(pcre).
-
-        's|'t|'re|'ve|'m|'ll|'d| ?\p{L}+| ?\p{N}+| ?[^\s\p{L}\p{N}]+|\s+(?!\S)|\s+
-        """
-        return torch.ops.torchtext.gpt2_bpe_pre_tokenizer(text)
-
-    @torch.jit.export
-    def _encode(self, text: str) -> List[int]:
-        """Encode text into a list of bpe token ids.
-
-        Split text into a list of token unit, and generate a list of bpe tokens
-        for each token unit. Lastly encode bpe tokens into bpe token ids.
-
-        Args:
-            text: An input text string.
-
-        Returns:
-            A list of bpe token ids represents each bpe tokens.
-
-        For example: "awesome,awe"
-            --> tokenize(regex) --> tokens: ["awesome", ",", "awe"]
-            --> bpe --> bpe tokens: ["aw", "esome"], [","], ["aw", e]
-            --> bpe encode --> bpe token ids: [707, 5927], [11], [707, 68]
-            --> result --> [707, 5927, 11, 707, 68]
-        """
-
-        bpe_token_ids: List[int] = []
-        for token in self._regex(text):
-            for bpe_token in self._bpe(self._byte_encode(token)):
-                bpe_token_ids.append(self.bpe_encoder[bpe_token])
-        return bpe_token_ids
+        self.bpe = torch.classes.torchtext.GPT2BPEEncoder(bpe_encoder, bpe_merge_ranks,
+                                                          self._seperator, bytes_to_unicode())
 
     @torch.jit.export
     def tokenize(self, text: str) -> List[str]:
```

Comment:
Also i missed this in previous PR, but I think converting token IDs into string is something we are doing internally to take care of vocab. I am not sure if this is a standard practice. To keep it consistent with Fairseq and OpenAI implementation, we should probably just return integer IDs. WDYT?

Reply:
I would suggest let's keep the ids as string for now till we standardize the tokenization API. My understanding is that the models that use this downstream expect strings as they build the vocab. Even the GPT-2 tokenizer for the model in HF transformers repo creates string ids.  [Reference code](https://github.com/huggingface/transformers/blob/master/src/transformers/models/gpt2/tokenization_gpt2.py#L252)